### PR TITLE
[Select] Improve spacing and types

### PR DESCRIPTION
**Pull request by FCalabria**
### WHY are these changes introduced?

There are two different changes here:

1. Set equal spacing to both size of the `prefix` when the select has an inline label.
Previously the spacing on the left side was smaller than the spacing on the right, which made the prefix look more related to the select label than to that specific option. Now both sides have equal spacing.
Before:

After:

2. Change type for `StrictOption.label` to allow the use of `React.ReactNode` too. 
This way we can use styled text on it

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

Spin: https://admin.web.select-changes.paqui-calabria.eu.spin.dev/store/shop1/collections/1
<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useCallback, useState} from 'react';
import {EnableSelectionMinor} from '@shopify/polaris-icons';

import {Icon, Page, Select, Text} from '../src';

export function Playground() {
  const [selected, setSelected] = useState('newestUpdate');

  const handleSelectChange = useCallback(
    (value: string) => setSelected(value),
    [],
  );

  const options = [
    {
      label: 'Better spacing',
      value: 'adjustSpacing',
      prefix: <Icon source={EnableSelectionMinor} />,
    },
    {
      label: (
        <Text as="span" color="success" fontWeight="bold">
          Label as react node
        </Text>
      ),
      value: 'reactNodeLabel',
    },
  ];
  return (
    <Page title="Playground">
      <Select
        label="Sort by"
        labelInline
        options={options}
        onChange={handleSelectChange}
        value={selected}
      />
    </Page>
  );
}

```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/happy-adults-accept.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Improve spacing and types on select component
```
Comment (laurkim) [inline .changeset/happy-adults-accept.md]:
```suggestion
Improved spacing and types on `Select` component
```

File: polaris-react/src/components/Select/Select.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ interface StrictOption {
   /** Machine value of the option; this is the value passed to `onChange` */
   value: string;
   /** Human-readable text for the option */
-  label: string;
+  label: string | React.ReactNode;
```
Comment (laurkim) [inline polaris-react/src/components/Select/Select.tsx]:
```suggestion
  label: React.ReactNode;
```
[ReactNode](https://reactnative.dev/docs/react-node) includes `string` as type so we don't need to explicitly include string.

**Comment by laurkim** [comment]
@FCalabria do you mind providing a spin instance that shows these changes? You can use the `/snapit` command to create a snapshot from this PR.

I'm also going to add @sarahill for review from a design perspective.

**Review by sarahill — APPROVED** [review]
✅

**Comment by FCalabria** [comment]
> @FCalabria do you mind providing a spin instance that shows these changes? You can use the `/snapit` command to create a snapshot from this PR.

@laurkim Sure! [Here you go](https://admin.web.select-changes.paqui-calabria.eu.spin.dev/store/shop1). I have changed the "best selling" option on the collection detail sort order so you can see both changes in action.

**Review by laurkim — APPROVED** [review]

**Pull request merged**